### PR TITLE
Deprecate @cbook.deprecated and move internal calls to @_api.deprecated

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

Followup of moving all API helpers to matplotlib._api (#18657).

### CODE REVIEW

File: lib/matplotlib/backends/_backend_pdf_ps.py [hunk 1]
```diff
@@ -27,7 +27,7 @@ class CharacterTracker:
     def __init__(self):
         self.used = {}
 
-    @mpl.cbook.deprecated("3.3")
+    @mpl._api.deprecated("3.3")
```
Comment (QuLogic) [inline lib/matplotlib/backends/_backend_pdf_ps.py:30]: Probably should import `_api` here instead.

File: lib/matplotlib/backends/qt_compat.py [hunk 2]
```diff
@@ -87,7 +87,7 @@ def _isdeleted(obj): return not shiboken2.isValid(obj)
         raise ValueError("Unexpected value for the 'backend.qt5' rcparam")
     _getSaveFileName = QtWidgets.QFileDialog.getSaveFileName
 
-    @mpl.cbook.deprecated("3.3", alternative="QtCore.qVersion()")
+    @mpl._api.deprecated("3.3", alternative="QtCore.qVersion()")
```
Comment (QuLogic) [inline lib/matplotlib/backends/qt_compat.py:90]: Ditto.

**Comment by QuLogic** [comment]
Test failures are relevant.

**Review by QuLogic — APPROVED** [review]

**Comment by timhoffm** [comment]
Doc builds fail because of
~~~
<unknown>:1: WARNING: py:class reference target not found: mpl_toolkits.axisartist.axisline_style._FancyAxislineStyle.FilledArrow
<unknown>:1: WARNING: py:class reference target not found: mpl_toolkits.axisartist.axisline_style._FancyAxislineStyle.SimpleArrow
~~~

Has anybody an idea where that comes from and how to fix it?

**Comment by QuLogic** [comment]
Going to ignore the doc failure, as it's not relevant to this PR.

**Comment by timhoffm** [comment]
Rebased to include #19163.

**Review by dopplershift — APPROVED** [review]

**Pull request merged**